### PR TITLE
[MPS] Fix 5D+ reductions over negative dimentions

### LINKED ISSUE #136132: torch.ones(2,4,1,30,1).to('mps').sum(dim=-2) throws "buffer is not large enough" on mps
### 🐛 Describe the bug

It doesn't happen with most shapes and dims, but on apple silicon:
```
python -c "import torch; torch.ones(2,4,1,30,1).to('mps').sum(dim=-2)"
```

throws:

```
/AppleInternal/Library/BuildRoots/ce725a5f-c761-11ee-a4ec-b6ef2fd8d87b/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:869: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 960 bytes
```

funny enough, 

```
python -c "import torch; torch.ones(2,4,1,30,1).to('mps').sum(dim=3)"
```

works just fine. `dim=-4` also throws the error, `dim=-3` doesn't. `torch.ones(2,4,1,30,2).to('mps').sum(dim=-2)` works too. Same behavior on python 3.12.6 and 3.10.14

### Versions

PyTorch version: 2.4.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.4 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.3.9.4)
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.6 | packaged by conda-forge | (main, Sep 11 2024, 04:55:15) [Clang 17.0.6 ] (64-bit runtime)
Python platform: macOS-14.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3 Pro

Versions of relevant libraries:
[pip3] numpy==2.1.1
[pip3] torch==2.4.1
[pip3] torchaudio==2.4.1
[pip3] torchvision==0.19.1
[conda] numpy                     2.1.1           py312h801f5e3_0    conda-forge
[conda] pytorch                   2.4.1                  py3.12_0    pytorch
[conda] torchaudio                2.4.1                 py312_cpu    pytorch
[conda] torchvision               0.19.1                py312_cpu    pytorch

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

### PR DESCRIPTION
**Pull request by malfet**
This fixes bug introduced by https://github.com/pytorch/pytorch/pull/99856 that attempts to speed-up reduction for 5D+ tensor if trailing dimensions are all ones, but introduces crashes/off-by-one errors for wrapped dimensions

Added regresion test case to `TestMPS.test_sum`

Fixes https://github.com/pytorch/pytorch/issues/136132

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!
